os/tools/compression/src/compress.c

```diff
@@ -65,41 +66,48 @@ static void compress_file(int block_size, int type, char *in_file, char *out_fil
 #if CONFIG_COMPRESSION_TYPE == LZMA
 	long unsigned int propsSize = LZMA_PROPS_SIZE;
 #endif
+	bool compression_success = true;
```

Comment:
How about having a variable named exit_code? Instead of setting the success value at each step, we can initialize the exit code to 2, and change it to 0 only if compression is success?

`int exit_code = 2;`
`...`
`/* Compression logic */`
`...`
`exit_code = 0`
`error:`
`exit(exit_code);`

Reply:
Thanks for the suggestion Kishore. I have modified the code accordingly.